### PR TITLE
Add serialization template to Schedule

### PR DESCRIPTION
**Pull request by akva2**
This adds serialization v2 for Schedule.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-simulators=2457 please

**Comment by akva2** [comment]
jenkins build this opm-simulators=2457 please

**Comment by akva2** [comment]
jenkins build this opm-simulators=2457 please

File: opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp [hunk 1]
```diff
@@ -418,6 +445,56 @@ namespace Opm
...
+            // to allow reconstructing the appropriate structures.
+            std::vector<std::pair<Key, std::vector<int>>> asMap;
+            std::vector<Type> unique;
+            for (const auto& it : map) {
+                for (const auto& w : it.second.data()) {
+                    if (std::find(unique.begin(), unique.end(), w) == unique.end())
+                        unique.push_back(w);
+                }
+            }
+            for (const auto& it : map) {
```
Comment (blattms) [inline opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp]: Somehow the second iteration over the map seems superfluous (and might be expensive due to the find). Is there a special reason not to do
```
std::vector<int> idxVec;

            for (const auto& it : map) {
                for (const auto& w : it.second.data()) {
                    auto candidate = std::find(unique.begin(), unique.end(), w);
                    auto idx = candidate - unique.begin();
                    if (candidate == unique.end())
                   {
                        unique.push_back(w);
                        idx = unique.size()-1;
                    }
                   idxVec.push_back(idx);
                }
                idxVec.push_back(it.second.initialRange());
                asMap.push_back(std::make_pair(it.first, idxVec));
            }

Comment (akva2) [reply opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp]: no reason, nice catch, done.

**Pull request merged**